### PR TITLE
🌱 Removes drain code copy from k8s

### LINKED ISSUE #3970: Remove forked kubectl drain code in favor of direct module usage
Previously we copied kubectl drain code into the third_party directory with the intention of eventually removing the copy and importing upstream directly (once the kubectl move to staging was complete). The kubectl code is now published at k8s.io/kubectl, however it appears that there have been changes made locally to the drain code we have copied.

These changes need to be re-aligned with upstream and we should be relying on an upstream import.

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
This patch removes the local copy of the drain code from the repo and instead uses the vendored code from the k8s.io/pkg/drain package.

**Which issue(s) this PR fixes**:
Fixes #3970

### CODE REVIEW

File: controllers/machine_controller.go [hunk 1]
```diff
@@ -534,21 +535,20 @@ func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cl
...
 
-	if err := kubedrain.RunCordonOrUncordon(ctx, drainer, node, true); err != nil {
+	if err := kubedrain.RunCordonOrUncordon(drainer, node, true); err != nil {
 		// Machine will be re-reconciled after a cordon failure.
 		log.Error(err, "Cordon failed")
 		return ctrl.Result{}, errors.Errorf("unable to cordon node %s: %v", node.Name, err)
 	}
 
-	if err := kubedrain.RunNodeDrain(ctx, drainer, node.Name); err != nil {
+	if err := kubedrain.RunNodeDrain(drainer, node.Name); err != nil {
```
Comment (vincepri) [inline controllers/machine_controller.go:551]: Is the context not yet upstream?

Comment (srm09) [reply controllers/machine_controller.go:551]: Didn't make the change for the other one :(

**Comment by srm09** [comment]
Might need to close this PR, looks like the ugly `kubernetes-drain` directory needs to stay here for a release longer. :(

File: controllers/machine_controller.go (hunk 1 above)
Comment (vincepri) [reply controllers/machine_controller.go:551]: Let's open another one, might be picked for next release, which is ok

**Comment by vincepri** [comment]
/close

Closing, per above message

File: controllers/machine_controller.go (hunk 1 above)
Comment (alexander-demichev) [reply controllers/machine_controller.go:551]: the context seems to be included in 1.21

Comment (srm09) [reply controllers/machine_controller.go:551]: The node drain portion was not using the context passed in the `DrainHelper` object. kubernetes/kubernetes#100439 uses the passed context for all the drain methods as well.

**Pull request closed without merging**